pkg/clusteragent/custommetrics/store.go

```diff
@@ -108,7 +108,7 @@ func NewConfigMapStore(client kubernetes.Interface, ns, name string) (Store, err
 }
 
 // SetExternalMetricValues updates the external metrics in the configmap.
-func (c *configMapStore) SetExternalMetricValues(added []ExternalMetricValue) error {
+func (c *configMapStore) SetExternalMetricValues(added map[string]ExternalMetricValue) error {
```

Comment:
in this method we should not do: `defer c.mu.Unlock()` like this, since the last line: `return c.updateConfigMap()` can take a while.

a solution can be to wrap the Map access (initialization and update) in a function
```
func() {
    c.mu.Lock()
	defer c.mu.Unlock()

	if c.cm == nil {
		return errNotInitialized
	}
	if c.cm.Data == nil {
		// Don't panic "assignment to entry in nil map" at init
		c.cm.Data = make(map[string]string)
	}
	for key, m := range added {
		toStore, err := json.Marshal(m)
		if err != nil {
			log.Debugf("Could not marshal the external metric %v: %v", m, err)
			continue
		}
		c.cm.Data[key] = string(toStore)
	}
}()
```